internal/logger/targets.go

```diff
@@ -39,8 +38,8 @@ type Target interface {
 }
 
 var (
-	// swapMu must be held while reading slice info or swapping targets or auditTargets.
-	swapMu sync.Mutex
+	swapAuditMuRW  sync.RWMutex
+	swapSystemMuRW sync.RWMutex
```

Comment:
Since locks are held so shortly for reads, regular mutexes are faster, since they don't have to deal with the housekeeping of RW mutextes.
```suggestion
	swapAuditMuRW  sync.Mutex
	swapSystemMuRW sync.Mutex
```
(plus adjust calls)

Reply:
How so @klauspost, we clearly don't want to compete on targets when concurrently logging.. it is better to never serialize such calls?

Reply:
@harshavardhana The read locks are held only while copying the 24 bytes of the slice. These operations are so fast they will never block for any significant time.

It does seem like they improved rwmutex to the point where the difference is insignificant. Writes block all reads anyway, so that will be the same.

Reply:
ref: https://github.com/golang/go/issues/38813